filebeat/beater/filebeat.go

```diff
@@ -98,8 +100,11 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 	// Start publisher
 	publisher.Start()
 	// Stopping publisher (might potentially drop items)
-	defer publisher.Stop()
-	defer successLogger.Close()
+	defer func() {
+		// Closes first the registrar logger to make sure not more events arrive at the registrar
```

Comment:
maybe add comment: registrarChannel must be closed first to potentially unblock (pretty unlikely) the publisher.